### PR TITLE
Fix #37 - reduce size of what we ship via dist/

### LINKED ISSUE #37: Reduce size of dist/
Right now the Brackets code in `dist/` (once we build it with `grunt build`) weighs in at 12.1M on the wire.  This includes the HTML, minified JS and CSS, images, fonts, etc.

Looking at the list of things we load, there are some obvious places to start:
* Get rid of any extensions we don't need from https://github.com/humphd/brackets/blob/bramble/src/utils/ExtensionLoader.js#L398-L420
* Don't build/ship source maps for JS and CSS (it's ~5M)
* Get rid of all the CodeMirror2 modes for languages we don't care about (they are all in `dist/thirdparty/CodeMirror2/mode/`).  We probably only care about html, css, js, markdown, json, and xml to start with.
* Investigate if there are other ways we can do our own custom build of CodeMirror2 to save space (i.e., those mode files aren't minified)
* Figure out which parts of Brackets itself we can dump.  For example, we don't need things like application auto-updating, installing new extensions, etc.

### PR DESCRIPTION
**Pull request by humphd**
This is a start, and gets us down to 3.4M (before gzipping, obviously).  I can make it better yet, but packaging up a customized, minified CodeMirror2 with a leaner set of modes.  I'll do that next week, I'm just trying to decide the right way to do it.

### CODE REVIEW

**Comment by KeniMorri** [comment]
r+ (I think thats how you use that?) tested, works fine

File: src/utils/ExtensionLoader.js [hunk 1]
```diff
@@ -401,24 +401,24 @@ define(function (require, exports, module) {
...
                     "QuickOpenHTML",
                     "QuickOpenJavaScript",
                     "QuickView",
-                    "RecentProjects",
+// XXXhumph: we don't yet support multiple projects, removing to save size
+//                  "RecentProjects",
                     "UrlCodeHints",
-                    "WebPlatformDocs",
+// XXXhumph: having this long term would be good, but removing for now to save size
+//                  "WebPlatformDocs",
```
Comment (peterflynn) [inline src/utils/ExtensionLoader.js]: Note that the WebPlatformDocs extension only loads its docs dataset lazily, the first time its invoked -- so including that big css.json file in dist won't affect initial Brackets load time at all.

Comment (humphd) [inline src/utils/ExtensionLoader.js]: Ah, good to know.  I assumed it was using require's text plugin to bundle.

File: Gruntfile.js [hunk 2]
```diff
@@ -64,18 +64,21 @@ module.exports = function (grunt) {
...
-                        dest: 'dist/',
-                        cwd: 'src/',
-                        src: [
-                            'extensibility/node/**',
-                            '!extensibility/node/spec/**',
-                            'filesystem/impls/appshell/node/**',
-                            '!filesystem/impls/appshell/node/spec/**'
-                        ]
-                    },
+//                    {
```
Comment (sedge) [inline Gruntfile.js]: Why is this commented vs deleted?

**Comment by sedge** [comment]
Why the commenting vs deleting?

**Comment by sedge** [comment]
Ahh, WIP?

**Comment by humphd** [comment]
Yeah, I was just experimenting.

File: src/utils/ExtensionLoader.js [hunk 3]
```diff
@@ -401,24 +401,18 @@ define(function (require, exports, module) {
                     "HtmlEntityCodeHints",
                     "InlineColorEditor",
                     "JavaScriptQuickEdit",
-                    "JSLint",
-                    "LESSSupport",
                     "QuickOpenCSS",
                     "QuickOpenHTML",
                     "QuickOpenJavaScript",
```
Comment (sedge) [inline src/utils/ExtensionLoader.js]: These `quickopen` modules... They're for quick editing other files from the context of the file that's loaded right? If we're only aiming at thimble, these can probably be removed for now.

File: src/utils/ExtensionLoader.js [hunk 4]
```diff
@@ -401,24 +401,18 @@ define(function (require, exports, module) {
...
-                    "RecentProjects",
                     "UrlCodeHints",
                     "WebPlatformDocs",
 
                     // Custom extensions we want loaded by default
-                    // NOTE: Maps to a folder inside /src/extensions/default/
                     "HTMLHinter",
                     "HideUI",
                     "thimbleProxy",
                     "brackets-browser-livedev"
```
Comment (sedge) [inline src/utils/ExtensionLoader.js]: Separate tangent, but `thimbleProxy` and `brackets-browser-livedev` probably should be combined at some point. *files*

Comment (humphd) [inline src/utils/ExtensionLoader.js]: Probably worth considering combining all our extensions into `brackets-browser-livedev`

**Comment by sedge** [comment]
Looks good to me! r+

**Pull request merged**